lib/gooddata/connection.rb

```diff
@@ -42,13 +42,13 @@ def disconnect
       Rest::Client.disconnect
     end
 
-    def with_connection(options = nil, second_options = nil, third_options = {}, &bl)
-      connection = connect(options, second_options, third_options)
-      bl.call(connection)
+    def with_connection(options = nil, second_options = nil, third_options = {}, &block)
+      client = connect(options, second_options, third_options)
+      block.call(client)
     rescue Exception => e # rubocop:disable RescueException
-      GoodData.logger.error(e.message)
       raise e
     ensure
+      client.connection.refresh_token
```

Comment:
This line means that the token will be refreshed always even when it's not expired. Can we rewrite it so that it only refreshes the token when it's needed?